samples/sample_common/src/vaapi_utils_drm.cpp

```diff
@@ -82,13 +82,34 @@ int open_intel_adapter(int type)
     return -1;
 }
 
-DRMLibVA::DRMLibVA(int type)
+int open_intel_adapter(int type, const char * devicePath)
+{
+    char driverName[MFX_DRM_DRIVER_NAME_LEN + 1] = {};
+
+    if(strcmp(devicePath,"") == 0)
+        return open_first_intel_adapter(type);
+
+    int fd = open(devicePath, O_RDWR);
+
+    if (fd < 0) {
+        close(fd);
```

Comment:
```return -1;``` ?

Reply:
wait. If fd is `<0` there is no need to call `close()` at the first place! Just need to return -1 right away. And I would print out an error that we failed to open the device.